src/test/java/org/assertj/core/api/object/ObjectAssert_hasNoNullFieldsOrProperties_Test.java

```diff
@@ -96,4 +96,17 @@ void should_pass_if_all_fields_or_properties_but_the_ignored_ones_are_set() {
     Jedi jedi = new Jedi("Yoda", null);
     assertThat(jedi).hasNoNullFieldsOrPropertiesExcept("lightSaberColor", "strangeNotReadablePrivateField");
   }
+
+  @Test
+  public void should_fail_if_a_field_does_not_exist() {
+    Jedi jedi = new Jedi("Yoda", "Blue");
+    assertThatExceptionOfType(AssertionError.class).isThrownBy(() -> assertThat(jedi).hasNoNullFieldsOrPropertiesExcept("father"))
+                                                   .withMessage(format("%n"
+                                                                       + "Expecting%n"
+                                                                       + "  org.assertj.core.test.Jedi%n"
+                                                                       + "to have the following declared fields:%n"
+                                                                       + "  [\"father\"]%n"
+                                                                       + "but it doesn't have:%n"
+                                                                       + "  [\"father\"]"));
```

Comment:
Use `expectAssertionError` as indicated in the [contributing guidelines](https://github.com/assertj/assertj-core/blob/main/CONTRIBUTING.md).